app/views/register/index.html.haml

```diff
@@ -3,30 +3,32 @@
 %header.page-header
   .grid-row
     .column-two-thirds
-      %h1.heading-large #{@register_name} Register
+      %h1.heading-large #{prepare_register_name(params[:register])} Register
     .column-one-third
-      = link_to "Add a new #{@register_name}", new_register_path, class: "button align-with-heading"
+      = link_to "Add a form #{prepare_register_name(params[:register])}", "#{params[:register]}/new", class: "button align-with-heading"
 
 .js-filter-list
   %form{class: "js-filter-form"}
     %fieldset
       .form-group
-        %label{class: "form-label", for: "search"} Search for a #{@register_name}
+        %label{class: "form-label", for: "search"} Search for a #{prepare_register_name(params[:register])}
         %input{name: "search", id: "search", class: "form-control"}
 
 %table
   %thead
     %tr
-      %th #{@register_name.capitalize}
-      %th Name
+      %th #{params[:register].capitalize}
+      - if @register[0].try(:name)
+        %th Name
       %th
 
   %tbody{class: "js-filter-block"}
     - @register.each do |register|
-      %tr{class: "js-filter-item", "data-filter-terms" => "#{register.name} #{register.key}"}
+      %tr{class: "js-filter-item", "data-filter-terms" => "#{register.name if register.try(:name)} #{register.key}"}
```

Comment:
`#{register.name if register.name}` will throw an exception because its a json object there is a possibility it is not there at all rather than just equal to nil.

`{register.try(:name)}` returns nil every time there is not something returned and html just renders nothing so I think thats OK, lets go with it. I made an assumption that try returns true or false and that nil would be rendered, this is cleaner and more succinct, good call.